nixos/modules/virtualisation/azure-agent.nix

```diff
@@ -146,15 +146,11 @@ in
 
     services.logrotate = {
       enable = true;
-      extraConfig = ''
-        /var/log/waagent.log {
-            compress
-            monthly
-            rotate 6
-            notifempty
-            missingok
-        }
-      '';
+      settings."/var/log/waagent.log" = {
+        compress = true;
```

Comment:
could this be a default?

Reply:
compress breaks some logs without delaycompress (if e.g. an application keeps writing to the just-rotated logs for a bit and we compress it we'll lose the last lines), so I think it makes more sense left as is.
If we also add delaycompress (would compress .2 files onwards) then I think it probably is ok to enable by default, but no other distro does this (see https://github.com/NixOS/nixpkgs/pull/162063#issuecomment-1060552102 for comparison between distros), so I'm a bit split on this.